cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -1592,36 +1593,54 @@ void reader::impl::allocate_columns(read_mode mode, size_t skip_rows, size_t num
     auto const d_cols_info = cudf::detail::make_device_uvector_async(
       h_cols_info, _stream, cudf::get_current_device_resource_ref());
 
+    cudf::detail::hostdevice_vector<size_t> sizes{_input_columns.size() * max_depth, _stream};
     auto const num_keys = _input_columns.size() * max_depth * subpass.pages.size();
+    auto constexpr max_keys_per_iter =
```

Comment:
Can we static cast this value as size_t? I think it might be helpful for safety in the expressions later on.

Reply:
Thanks for the early feedback, Bradley. Yes, that makes more sense!

Reply:
(I realized during/after my review that this was in-progress and maybe not ready for review. I got a review request because the PR was opened and then converted to draft.)

Reply:
No worries Bradley. Thank you!